frontend/src/Main.purs

```diff
@@ -33,7 +35,8 @@ config state = do
   -- socket
   actionChannel <- channel $ Ex.SocketConnected false
   let socketSignal = subscribe actionChannel :: Signal Ex.Action
-  socket' <- connect Ex.socketHost
+  let socketHost = Ex.mkSocketHost (protocol isProduction) (unsafePerformEff hostname)
```

Comment:
Why do we use `unsafePerformErr` here when we are already in `Eff`. Why not do it like:
```
  socketHost <- ExMkSocketHost (protocol isProduction) <$> hostname
```
as said before, much better would be even:
```
  let prodDevProtocol = if isProduction then Https else Http
  socketHost <- ExMkSocketHost (protocolToString prodDevProtocol) <$> hostname
```